src/contexts/PokemonDataContext.js

```diff
@@ -19,33 +25,75 @@ export const PokemonDataProvider = ({ children }) => {
     }
   }, [currentPokemonId]);
 
+  useEffect(() => {
+    if (filters) {
+      filterPokemons();
+    } else {
+      setPokemons(allPokemons);
+    }
+    setRefresh((item) => !item);
+  }, [filters]);
+
```

Comment:
According to the if-else statemnt in your filterPokemons() function, if filters is null, setPokemons(allPokemons) is executed, so you rather don't need another if-else statement here.

```suggestion
  useEffect(() => {
    filterPokemons();
    setRefresh((item) => !item);
  }, [filters]);

```